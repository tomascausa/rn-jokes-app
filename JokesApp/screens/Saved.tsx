import React, {useContext} from 'react';
import {StyleSheet, SafeAreaView, FlatList} from 'react-native';
import {Context} from '../store/context';
import {Joke} from '../types/Joke';
import JokeCard from '../components/UI/JokeCard';
import Separator from '../components/Carousel/Separator';

const Saved: React.FC = () => {
    const context = useContext(Context);

    return (
        <SafeAreaView style={styles.container}>
            <FlratList
                data={context.jokes}
                numColumns={2}
                showsVerticalScrollIndicator={false}
                ItemSeparatorComponent={Separator}
            />
        </SafeAreaView>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'column',
        marginTop: 40,
        paddingHorizontal: 15,
    },
});

export default Saved;
