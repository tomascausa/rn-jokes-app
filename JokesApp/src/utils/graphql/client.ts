import {ApolloClient, InMemoryCache} from '@apollo/client';

export const client = new ApolloClient({
    uri: 'https://icanhazdadjoke.com/graphql',
    cache: new InMemoryCache(),
});
