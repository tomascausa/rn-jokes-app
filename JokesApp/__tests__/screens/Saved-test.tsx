import React from 'react';
import {MockedProvider} from '@apollo/client/testing';
import {NavigationContainer} from '@react-navigation/native';
import renderer from 'react-test-renderer';
import Saved from '../../src/screens/Saved';

describe('<Saved />', () => {
    test('renders correctly', () => {
        const component = (
            <MockedProvider>
                <NavigationContainer>
                    <Saved />
                </NavigationContainer>
            </MockedProvider>
        );
        const json = renderer.create(component).toJSON();
        expect(json).toMatchSnapshot();
    });
});
