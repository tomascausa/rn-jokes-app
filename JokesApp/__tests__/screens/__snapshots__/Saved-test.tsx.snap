// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Saved /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flex": 1,
        "marginHorizontal": 20,
        "marginTop": 30,
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "Roboto-Medium",
        "fontSize": 25,
        "textAlign": "center",
      }
    }
  >
    No saved jokes found
  </Text>
</RCTSafeAreaView>
`;
